clang/lib/Sema/SemaStmtAttr.cpp

```diff
@@ -322,6 +322,66 @@ static Attr *handleUnlikely(Sema &S, Stmt *St, const ParsedAttr &A,
   return ::new (S.Context) UnlikelyAttr(S.Context, A);
 }
 
+CodeAlignAttr *Sema::BuildCodeAlignAttr(const AttributeCommonInfo &CI,
+                                        Expr *E) {
+  if (!E->isValueDependent()) {
+    llvm::APSInt ArgVal;
+    ExprResult Res = VerifyIntegerConstantExpression(E, &ArgVal);
+    if (Res.isInvalid())
+      return nullptr;
+    E = Res.get();
+
+    // This attribute requires an integer argument which is a constant power of
+    // two between 1 and 4096 inclusive.
+    int AlignValue = ArgVal.getSExtValue();
```

Comment:
this should be int64.  Also, we probably  need a test for 'larger than 64 bit' I think?  Else this is going to give weird behavior.

Reply:
To be more clear, I think SExt makes sense, but this would cause a problem for `9223372036854775808` (int64-max +1).

Reply:
with SExt, we can see this:
``
test/Sema/code_align.c:69:5: error: 'code_align' attribute requires an integer argument which is a constant power of two between 1 and 4096 inclusive; got -9223372036854775808
    [[clang::code_align(9223372036854775808)]]
       ^
``
With ZExt, we can see correct value:
```
uint64_t AlignValue = ArgVal.getZExtValue();
test/Sema/code_align.c:69:5: error: 'code_align' attribute requires an integer argument which is a constant power of two between 1 and 4096 inclusive; got 9223372036854775808
     [[clang::code_align(9223372036854775808)]]
````

Reply:
> I think SExt makes sense, but this would cause a problem for `9223372036854775808` (int64-max +1).

Yes.

@erichkeane,  I have updated patch to avoid this kind of problem with commit https://github.com/llvm/llvm-project/pull/70762/commits/088767463be990eb8af4e62a6c66cf4ba06613c3.